data-prepper-plugin-framework/src/main/java/org/opensearch/dataprepper/plugin/PluginBeanFactoryProvider.java

```diff
@@ -61,4 +63,18 @@ public BeanFactory get() {
         final GenericApplicationContext isolatedPluginApplicationContext = new GenericApplicationContext(sharedPluginApplicationContext);
         return isolatedPluginApplicationContext.getBeanFactory();
     }
+
+    public BeanFactory initializePluginSpecificIsolatedContext(Class[] markersToScanForDI) {
+        AnnotationConfigApplicationContext pluginDIContext = new AnnotationConfigApplicationContext();
+        if(markersToScanForDI.length>0) {
```

Comment:
Do we need to check for non-null here?

Reply:
Added null check now. From the annotation, we have the default value of empty array but still someone could call this method with a null value so better to have the check. I added it now. Thanks for catching it.